### DISCUSSION TITLE
What's the difference between torch_geometric.sampler.NeighborSampler and torch_geometric.loader.NeighborSampler?

### DISCUSSION THREAD

**Discussion by jeyendranbalakrishnan** [pyg-team/pytorch_geometric#7423] [General]
- Which one should I use?
- It appears that torch_geometric.sampler.NeighborSampler is used by torch_geometric.loader.NeighborLoader. Is this correct?
Thanks a lot!

**Comment by rusty1s** [comment]
* You should use `NeighborLoader` - `loader.NeighborSampler` is deprecated and will be removed soon.
* Yes, `NeighborSampler` is an internal class in `NeighborLoader` that is used for graph sampling (and calls the specific C++ low-level routines).

**Reply by jeyendranbalakrishnan** [reply]
Thanks a lot! That helps me understand better.
One follow up question:
* Can we view NeighborLoader as a sort of "improved" version of DataLoader that adds neighbor subsampling capability for nodes in a minibatch, whereas DataLoader copies all the neighbors into each minibatch?

**Reply by rusty1s** [reply]
Mh, no, they serve different purposes. `NeighborLoader` enables mini-batch sampling on single large graphs (e.g., for learning on a single social network), while `DataLoader` corresponds to the traditional scenario of grouping multiple (small) graphs into a mini-batch (e.g., for learning on a large dataset holding multiple molecules).

**Reply by jeyendranbalakrishnan** [reply]
Thanks for the clarification, @rusty1s.
